src/v/cluster/partition.cc

```diff
@@ -124,6 +125,13 @@ partition::partition(
                     cloud_storage_clients::bucket_name{*bucket});
             }
         }
+
+        // Construct cloud_storage write path (ntp_archiver)
+        if (
+          config::shard_local_cfg().cloud_storage_enabled()
+          && cloud_storage_api.local_is_initialized()
+          && _raft->ntp().ns == model::kafka_namespace) {
```

Comment:
looks like the branch is empty

Reply:
I think it gets filled out in a subsequent commit.